api/server/router/swarm/cluster_routes.go

```diff
@@ -270,6 +280,11 @@ func (sr *swarmRouter) updateService(ctx context.Context, w http.ResponseWriter,
 				// Sysctls for docker swarm services weren't supported before
 				// API version 1.40
 				service.TaskTemplate.ContainerSpec.Sysctls = nil
+
+				if service.TaskTemplate.ContainerSpec.Privileges != nil && service.TaskTemplate.ContainerSpec.Privileges.CredentialSpec != nil {
+					// Support for setting credential-spec through configs was added in API 1.40
+					service.TaskTemplate.ContainerSpec.Privileges.CredentialSpec.Config = ""
+				}
```

Comment:
I guess we need to strip the `Runtime` property here as well (as is done in `createService`)?
```suggestion
				}
				for _, config := range service.TaskTemplate.ContainerSpec.Configs {
					// support for the Runtime target was added in API 1.40
					config.Runtime = nil
				}
```